server/llm/openai_assistant.py

```diff
@@ -61,10 +64,11 @@ def query(self, custom_query: str = None) -> str:
                 temperature=0,
             )
             for choice in res.choices:
-                if choice.finish_reason == "stop":
+                reason = choice.finish_reason
+                if reason == "stop" or reason == "length":
```

Comment:
Adding "length" in case we specify max output tokens at some point (which I have experimented with but opted to leave out for now. Another option would be to specify max output length only for the default query, not custom.